AirCasting/Dashboard/EmptyDashboardView.swift

```diff
@@ -10,15 +10,25 @@ import SwiftUI
 
 struct EmptyDashboardView: View {
     @EnvironmentObject private var tabSelection: TabBarSelection
+    @EnvironmentObject var selectedSection: SelectSection
+    @EnvironmentObject var sessionSynchronizationController: SessionSynchronizationController
     var body: some View {
         emptyState
     }
+    var shouldSessionFetch: Bool {
+        (selectedSection.selectedSection == .mobileDormant || selectedSection.selectedSection == .fixed) && sessionSynchronizationController.syncInProgress
```

Comment:
I think it would be better to keep `SessionSynchronizationController` as a non-ObservableObject entity and create a ViewModel that in turn will be a proper `ObservableObject` and will contain all this UI logic. Two reasons:
* any time the synchronization state changes it will currently cause to reload basically all apps UI since it's used by `AirCastingApp.swift`. We can avoid that and only reload the UI when and where it's actually needed by creating a VM
* code clarity - we move out the logic part of the UI to a special-purposed class (VM) which makes `View` only about layout as it should.
* mitigating conflicts - any logic put inside views will generate lots of conflicts for us as all departments are meeting in the same files instead of being separated. We're experiencing this already a little.